daft/filesystem.py

```diff
@@ -154,10 +161,10 @@ def _resolve_paths_and_filesystem(
     if resolved_filesystem is None:
         # Resolve path and filesystem for the first path.
         # We use this first resolved filesystem for validation on all other paths.
-        resolved_path, resolved_filesystem = _infer_filesystem(paths[0], io_config)
+        resolved_path, resolved_filesystem, expiry = _infer_filesystem(paths[0], io_config)
```

Comment:
Could be cleaner instead if we wrapped the filesystem and expiry into a dataclass (e.g. `PyarrowFilesystemWithExpiry`)?

Reply:
Yeah that sounds good, I'll do that